src/Utils/citesupport.ts

```diff
@@ -0,0 +1,453 @@
+"use strict";
+import CiteWorker from "./cite.worker.ts";
+/* global Word OfficeExtension*/
+
+class CiteSupport {
+  config: {
+    debug: boolean;
+    mode: string;
+    defaultLocale: string;
+    defaultStyle: string;
+    citationIdToPos: {};
+    citationByIndex: object[];
+    processorReady: boolean;
+    citationData: object[];
+  };
+  worker: Worker;
+  constructor(citationData: object[]) {
+    this.config = {
+      debug: true,
+      mode: "in-text",
+      defaultLocale: "en-US",
+      defaultStyle: "american-sociological-association",
+      citationIdToPos: {},
+      citationByIndex: [],
+      processorReady: false,
+      citationData: citationData,
+    };
+    var me = this;
+    this.worker = new CiteWorker();
+    this.worker.onmessage = function (e) {
+      switch (e.data.command) {
+        /**
+         *    In response to `callInitProcessor` request, refresh
+         *   `config.mode`, and document citations (if any)
+         *    and document bibliography (if any).
+         *
+         * @param {string} xclass Either `note` or `in-text` as a string
```

Comment:
It's really ugly, that the interface between the worker and the caller is defined via these comments instead of a proper typescript interface. In a different context, I'd implemented a nice typed interface for message transfer here: https://github.com/Zettlr/Zettlr/pull/1048 Maybe one can reuse some of these ideas to define a strongly typed worker? I did a quick google search and found https://aless.co/typed-web-workers/ which looks promising (but I'm unsure if it gives also typed access to the `data` object.

At the very least, it should be possible to hide the not typed part in a simple façade, which then for example exposes `onInitProcessor(xclass, citationByIndex, rebuildData)` and takes care of how this data is extracted from the worker's event.